### PR TITLE
[web] adding firefox unit tests to font loading

### PR DESCRIPTION
**Pull request by nturgut**
flutter web engine unit tests is running on firefox (as an addition to chrome) with PR: flutter/engine#14267

The following tests were skipped on Firefox since they were failing:
test/text/font_collection_test.dart: FontManager Register Asset twice with special character slash
test/text/font_collection_test.dart: FontManager Register Asset twice with exclamation mark
test/text/font_collection_test.dart: FontManager Register Asset twice with coma
test/text/font_collection_test.dart: FontManager Register Asset twice with a digit at the start of a token

The failure was legitimate since the loading was handled differently for Firefox. Added a new group of tests to test Firefox and skipped the other group.

Related: https://github.com/flutter/flutter/issues/46638

### CODE REVIEW

File: lib/web_ui/test/text/font_collection_test.dart [hunk 1]
```diff
@@ -21,122 +21,192 @@ void main() {
...
+        expect(fontFamilyList.length, equals(2));
+        expect(fontFamilyList, contains('\'Ahem!!ahem\''));
+        expect(fontFamilyList, contains('Ahem!!ahem'));
       });
 
-      expect(fontFamilyList.length, equals(2));
-      expect(fontFamilyList, contains('\'/Ahem\''));
-      expect(fontFamilyList, contains('/Ahem'));
-    }, skip: (browserEngine == BrowserEngine.firefox));
+      test('Register Asset twice with coma', () async {
```
Comment (hterkelsen) [inline lib/web_ui/test/text/font_collection_test.dart]:
```suggestion
      test('Register Asset twice with comma', () async {
```

File: lib/web_ui/test/text/font_collection_test.dart [hunk 2]
```diff
@@ -21,122 +21,192 @@ void main() {
...
-      final String testFontFamily = 'Ahem 1998';
-      final List<String> fontFamilyList = List<String>();
-
-      fontManager.registerAsset(
-          testFontFamily, 'url($_testFontUrl)', const <String, String>{});
-      await fontManager.ensureFontsLoaded();
-      html.document.fonts
-          .forEach((html.FontFace f, html.FontFace f2, html.FontFaceSet s) {
-        fontFamilyList.add(f.family);
+      test('Register Asset with coma', () async {
```
Comment (hterkelsen) [inline lib/web_ui/test/text/font_collection_test.dart]:
```suggestion
      test('Register Asset with comma', () async {
```

File: lib/web_ui/test/text/font_collection_test.dart (hunk 1 above)
Comment (nturgut) [reply lib/web_ui/test/text/font_collection_test.dart]: thanks :) done.

File: lib/web_ui/test/text/font_collection_test.dart (hunk 2 above)
Comment (nturgut) [reply lib/web_ui/test/text/font_collection_test.dart]: done.

**Comment by nturgut** [comment]
Merging this one. This file is only adding more unit tests, no change in code behavior.

File: lib/web_ui/test/text/font_collection_test.dart [hunk 3]
```diff
@@ -21,122 +21,192 @@ void main() {
...
+        fontManager.registerAsset(
+            _testFontFamily, 'url($_testFontUrl)', const <String, String>{});
+        await fontManager.ensureFontsLoaded();
+        html.document.fonts
+            .forEach((html.FontFace f, html.FontFace f2, html.FontFaceSet s) {
+          fontFamilyList.add(f.family);
+        });
+
+        expect(fontFamilyList.length, equals(1));
+        expect(fontFamilyList.first, '\"/Ahem\"');
```
Comment (yjbanov) [inline lib/web_ui/test/text/font_collection_test.dart]: Are the expectations the only difference between this test and the non-Firefox? If so, I think I'd prefer that we write one test and use different `expect`s, e.g.:

```dart
test('...', () async {
  ... shared test code ...
  if (browserEngine == firefox) {
    ... FF expectations ...
  } else {
    ... other expectations ...
  }
});
```

This way we won't forget to run tests in some browsers.

Comment (nturgut) [reply lib/web_ui/test/text/font_collection_test.dart]: Thanks, I merged the PR in the early morning. I'll address this in with another PR.

Comment (nturgut) [reply lib/web_ui/test/text/font_collection_test.dart]: sent this one out: https://github.com/flutter/engine/pull/14487

**Pull request merged**